src/vm/appdomain.cpp

```diff
@@ -12583,6 +12583,8 @@ AppDomain::RaiseAssemblyResolveEvent(
             if ( pSpec->IsIntrospectionOnly() 
 #ifdef FEATURE_FUSION
                     || pSpec->GetParentLoadContext() == LOADCTX_TYPE_UNKNOWN
+#elif FEATURE_CORECLR
+                    || pSpec->GetBindingContextFromParentAssembly(this) != GetTPABinderContext()
 #endif
```

Comment:
It may be better to not try to be smart here and just set it for contexts for consistency. I know that it is different from desktop, but it will make it easier for people to make things diagnose and fix when they run into issues.